drivers/gpio/gpio_ambiq.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2023 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ambiq_gpio
+
+#include <errno.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/irq.h>
+#include <zephyr/spinlock.h>
+
+#include <am_mcu_apollo.h>
+
+#define BITS 32
```

Comment:
you're likely looking for `gpio-controller.yaml` `ngpios` instead?

Reply:
Exactly. Thanks!